_posts/2019-03-04-jumping-on-the-dash-less-bandwagon.markdown

```diff
@@ -0,0 +1,64 @@
+
+Serverless grants unrivalled visibility into the cost and profitability of your business at an incredibly fine-grained level. You can literally see how much it costs to provide the ‘Contact us’ form on your website. Don’t believe me? Read the thought-provoking article at the end of this section.
+
+Serverless agility is about the highlighting (and hopefully the removal) of waste. Anything that gets in the way of delivering value is immediately highlighted, so if it is a technical hurdle then go around it; if its operational then work out how to kill it or change it. End customers don’t care about what bureaucracy you think you need, they’ll vote with eyeballs and wallets and just go somewhere else if you can’t deliver the goods.
+
+**Link of Interest:** [Micro-Optimization: Activity-Based Costing for Digital Services?](https://aws.amazon.com/blogs/enterprise-strategy/micro-optimization-activity-based-costing-for-digital-services/)
+
+## Technical Impacts
+
+For all you technical readers out there, be afraid of serverless. It isn’t a flash in the pan buzzword that’s going away. It is coming for you and will make you obsolete if you don’t adapt. The full power of serverless requires a different way of working and thinking.
+
+Firstly, it removes the infrastructure downtime between tasks. In the past you’d have to wait for servers to become available. Then you need to wait for software (like application servers) to be installed and configured. Then it was gaining database access for your code. Etcetera etcetera etcetera. With serverless, once you have your access permissions, you can have a brand-new application, in production, within hours. Everything needed to host and operate your application is provided to you by the cloud provider.
+
+Next, with all this infrastructure provided to you, gone are hours or days of writing boilerplate code. All the lambda functions I’ve written listen for an incoming event (http, file, database, whatever) and consume it. No setting up scaffolding code. No writing complex convoluted setup logic for unit tests. Just the minimum code to consume the event and suite of testing scenarios to verify the behaviours that I expect. [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) at its finest.
+
+Time to think in terms of events and states as ultimately you’re responding to state change events within the bounds of a fully distributed system. This requires a more disciplined focus on what triggers these events and how things can, and will, [go awry at any time](https://en.wikipedia.org/wiki/Chaos_engineering).
+
+The architecture diagram below for a photo management service, on a recent project, demonstrates just how powerful Serverless can be. The service allows end-users to take, upload and manage maintenance photos of the organisation's physical equipment in the field. To deliver this, nothing more beyond CloudFormation scripts and Lambda code was needed.
```

Comment:
In this para at the end, you're assuming AWS (cfn, lambda) which is fine, but the rest of the post, you've tried to be agnostic. You may want to rephrase eg:
> nothing more beyond infrastructure scripts, (eg terraform, cloudformation) and your function code (lambda, google/azure functions)

Reply:
or even just:
> and your function code.

Reply:
Or: 
> Infrastructure as Code [link to wikipedia, or cloudformation] and application logic

Reply:
Reworded to the below. Since the diagram itself is AWS-centric I don't think making the words agnostic makes sense. I do think though the flow was subpar 😃 

> A recent project demonstrates just how powerful Serverless can be. The AWS Architecture Diagram below is for a photo management service. The service allows end-users to take, upload and manage maintenance photos of the organisation's physical equipment in the field. To deliver this, nothing more beyond CloudFormation scripts and Lambda code was needed.